google-cloud-spanner/lib/google/cloud/spanner/fields.rb

```diff
@@ -0,0 +1,175 @@
+# Copyright 2017 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+require "google/cloud/spanner/convert"
+
+module Google
+  module Cloud
+    module Spanner
+      ##
+      # # Fields
+      #
+      # ...
+      class Fields
+        ##
+        # @private
+        def initialize types
+          @fields = []
+          if types.is_a? Array
+            types.each do |type|
+              @fields << field(type)
+            end
+          elsif types.is_a? Hash
+            types.each do |type|
+              @fields << field(type)
```

Comment:
They are. Both Array and Hash use the same API. Originally those lines were not identical, but they were refactored and are the same now. They may diverge again in the future, so I felt it was best to keep them as they are for now.